File: tlb/09/solutions/distinct_lines.v1.py
#!/usr/bin/python3
# written by evank@cse.unsw.edu.au

import sys
import re
n = int(sys.argv[1])
seen = set()
for i, line in enumerate(sys.stdin, start=1):
    seen.add(re.sub(r"\s", "", line.lower())
    if len(seen) == n:
        print("{} distinct lines seen after {} lines read.".format(len(seen), i)
        break
else:
    print("End of input reached after {} lines read - {} different lines not seen.".format(i, n))

File: tlb/07/solutions/distinct_lines.v1.pl
#!/usr/bin/perl -w

$n_distinct_lines_needed = shift @ARGV or die;

$n_lines = 0;
while ($line = <STDIN>) {
	$n_lines++;
	$line = lc $line;
	$line =~ s/\s//g;
	$lines_seen{$line}++;
	if (keys %lines_seen >= $n_distinct_lines_needed) {
		print "$n_distinct_lines_needed distinct lines seen after $n_lines lines read.\n";
		exit 0;
	}
}
print "End of input reached after $n_lines lines read - $n_distinct_lines_needed different lines not seen.\n";
